src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/TerminalPane.java

```diff
@@ -949,7 +949,30 @@ private void setFocusOnVisible()
       if (visibleTerminal != null)
       {
          if (!suppressAutoFocus_)
-            Scheduler.get().scheduleDeferred(() -> visibleTerminal.setFocus(true));
+            Scheduler.get().scheduleDeferred(() -> {
+               // On rare occasions (which I haven't been able to nail down), flow gets here
+               // before xtermjs has initialized, and the setFocus call throws a null exception.
+               // Nothing is obviously broken when that happens, but guard against it, wait
+               // a tiny bit and try once more.
+               if (visibleTerminal.terminalEmulatorLoaded())
+               {
+                  visibleTerminal.setFocus(true);
+               }
+               else
+               {
+                  new Timer()
```

Comment:
FYI, the recently-added `Timers.singleShot(200, () -> { ... })` would be a more succinct way of expressing this. https://github.com/rstudio/rstudio/blob/c74f6f35996dca9e1572e3f647549e96bf0c0e19/src/gwt/src/org/rstudio/studio/client/common/Timers.java#L22-L32

Reply:
Nice!